### PR TITLE
#1 PersonService

### PR DESCRIPTION
**Pull request by smirnovV**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+        </dependency>
+
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-web</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>com.h2database</groupId>
+            <artifactId>h2</artifactId>
```
Comment (01sas) [inline pom.xml]: Зачем?

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    <dependencies>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-data-jpa</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>postgresql</groupId>
+            <artifactId>postgresql</artifactId>
+            <version>9.1-901-1.jdbc4</version>
```
Comment (01sas) [inline pom.xml]: Старая версия

File: src/main/java/ru/smirnovv/Application.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    public static void main(final String[] args) {
+        SpringApplication.run(Application.class, args);
+    }
+
+    /**
+     * Class that handles global exception.
+     */
+    @ControllerAdvice
+    public static class GlobalExceptionHandler {
+
```
Comment (01sas) [inline src/main/java/ru/smirnovv/Application.java]: Убери строку

File: src/main/java/ru/smirnovv/ErrorType.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
+package ru.smirnovv;
+
+/**
+ * Special class that contains error information.
+ */
+public class ErrorType {
+
```
Comment (01sas) [inline src/main/java/ru/smirnovv/ErrorType.java]: Убери строку

File: src/main/java/ru/smirnovv/person/InvalidNameException.java [hunk 5]
```diff
@@ -0,0 +1,16 @@
+package ru.smirnovv.person;
+
+/**
+ * Exception indicates that the name does not conform to the naming syntax.
+ */
+public class InvalidNameException extends RuntimeException {
+
```
Comment (01sas) [inline src/main/java/ru/smirnovv/person/InvalidNameException.java]: Убери строку

File: src/main/java/ru/smirnovv/person/Person.java [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+import javax.validation.constraints.Pattern;
+import javax.validation.constraints.Size;
+
+/**
+ * Entity person containing id and name.
+ */
+@SuppressWarnings("magicNumber")
+@Entity
+public class Person {
+
```
Comment (01sas) [inline src/main/java/ru/smirnovv/person/Person.java]: Убери строку

File: src/main/java/ru/smirnovv/person/Person.java [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+    /**
+     * Returns persons id.
+     *
+     * @return id.
+     */
+    public final Long getId() {
+        return id;
+    }
+
```
Comment (01sas) [inline src/main/java/ru/smirnovv/person/Person.java]: Убери строку

File: src/main/java/ru/smirnovv/person/PersonController.java [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+import static org.springframework.util.Assert.notNull;
+import static ru.smirnovv.person.PersonService.checkName;
+
+/**
+ * Class handles requests associated with a list of person.
+ */
+@RestController
+@RequestMapping("/person")
+public class PersonController {
+
```
Comment (01sas) [inline src/main/java/ru/smirnovv/person/PersonController.java]: Убери строку

File: src/main/java/ru/smirnovv/person/PersonService.java [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.util.Assert;
+
+/**
+ * Service that manages list of person.
+ */
+@SuppressWarnings({"designForExtension", "magicNumber"})
+@Service
+public class PersonService {
+
```
Comment (01sas) [inline src/main/java/ru/smirnovv/person/PersonService.java]: Убери строку

File: src/test/java/ru/smirnovv/person/PersonControllerTest.java [hunk 10]
```diff
@@ -0,0 +1,265 @@
...
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+@AutoConfigureMockMvc
+public class PersonControllerTest {
```
Comment (01sas) [inline src/test/java/ru/smirnovv/person/PersonControllerTest.java]: Добавь комментарий

File: .gitignore [hunk 11]
```diff
@@ -0,0 +1,4 @@
+.idea/
```
Comment (smirnovV) [inline .gitignore]: test

File: src/main/resources/application.yml [hunk 12]
```diff
@@ -0,0 +1,10 @@
+spring:
+  datasource:
+    username: slava
+    password: andromeda
+    url: jdbc:postgresql://localhost:5432/iris_db
+  jpa:
+    database: POSTGRESQL
+    generate-ddl: true
+    hibernate.ddl-auto: update
```
Comment (01sas) [inline src/main/resources/application.yml]: validate

File: src/main/java/ru/smirnovv/Application.java [hunk 13]
```diff
@@ -0,0 +1,85 @@
...
+import static org.springframework.http.HttpStatus.NOT_FOUND;
+
+/**
+ * Service that search people by iris.
+ */
+@SuppressWarnings("hideUtilityClassConstructor")
+@SpringBootApplication
+public class Application {
+    /**
+     * The entry point where loads spring Application.
```
Comment (01sas) [inline src/main/java/ru/smirnovv/Application.java]: The entry point that bootstraps application.

File: src/main/java/ru/smirnovv/Application.java [hunk 14]
```diff
@@ -0,0 +1,85 @@
...
+
+    /**
+     * The class that handles the global exception.
+     */
+    @ControllerAdvice
+    public static class GlobalExceptionHandler {
+        /**
+         * Handles {@link NotFoundException} and returns the response with error information.
+         *
+         * @param request   request where thrown exception.
```
Comment (01sas) [inline src/main/java/ru/smirnovv/Application.java]: the request where the exception was thrown.

File: src/main/java/ru/smirnovv/Application.java [hunk 15]
```diff
@@ -0,0 +1,85 @@
...
+         * @param exception thrown {@link NotFoundException}.
```
Comment (01sas) [inline src/main/java/ru/smirnovv/Application.java]: the thrown exception.

File: src/main/java/ru/smirnovv/Application.java [hunk 16]
```diff
@@ -0,0 +1,85 @@
...
+                    INTERNAL_SERVER_ERROR);
+        }
+
+        /**
+         * Handles {@link MissingServletRequestParameterException} and
+         * returns the response with error information.
+         *
+         * @param request   request where thrown exception.
+         * @param exception thrown {@link MissingServletRequestParameterException}.
+         * @return responseEntity with error information.
```
Comment (01sas) [inline src/main/java/ru/smirnovv/Application.java]: a response entity with error details.

File: src/main/java/ru/smirnovv/Application.java [hunk 17]
```diff
@@ -0,0 +1,85 @@
...
+         * @see ErrorType
+         */
+        @ExceptionHandler(MissingServletRequestParameterException.class)
+        public final ResponseEntity<?> missingRequestParameterException(
```
Comment (01sas) [inline src/main/java/ru/smirnovv/Application.java]: Specify exact type instead of ?

File: src/main/java/ru/smirnovv/ErrorType.java [hunk 18]
```diff
@@ -0,0 +1,62 @@
...
+     */
+    private final int status;
+
+    /**
+     * Message explanation of the problem.
+     */
+    private final String message;
+
+    /**
+     * Constructs an instance of ErrorType using an URL,
```
Comment (01sas) [inline src/main/java/ru/smirnovv/ErrorType.java]: Constructs an instance with the specified properties.

File: src/main/java/ru/smirnovv/NotFoundException.java [hunk 19]
```diff
@@ -0,0 +1,15 @@
+package ru.smirnovv;
+
+/**
+ * This exception indicates that the requested resource was not found.
+ */
+public class NotFoundException extends RuntimeException {
+    /**
+     * Constructs an instance of NotFoundException using an explanation of the problem.
```
Comment (01sas) [inline src/main/java/ru/smirnovv/NotFoundException.java]: Constructs an instance with the specified detail message.

File: src/main/java/ru/smirnovv/NotFoundException.java [hunk 20]
```diff
@@ -0,0 +1,15 @@
...
+     *
+     * @param s message explanation of the problem.
```
Comment (01sas) [inline src/main/java/ru/smirnovv/NotFoundException.java]: @param message the detail message.

File: src/main/java/ru/smirnovv/person/InvalidNameException.java [hunk 21]
```diff
@@ -0,0 +1,15 @@
+package ru.smirnovv.person;
+
+/**
+ * This exception indicates that the name does not conform to the naming syntax.
```
Comment (01sas) [inline src/main/java/ru/smirnovv/person/InvalidNameException.java:4]: Please specify restrictions.

File: src/main/java/ru/smirnovv/person/Person.java [hunk 22]
```diff
@@ -0,0 +1,74 @@
...
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.Pattern;
+import javax.validation.constraints.Size;
+
+/**
+ * Entity person containing id and name.
```
Comment (01sas) [inline src/main/java/ru/smirnovv/person/Person.java]: An entry that represents a registered in the system person.

File: src/main/java/ru/smirnovv/person/Person.java [hunk 23]
```diff
@@ -0,0 +1,74 @@
...
+ */
+@SuppressWarnings("magicNumber")
+@Entity
+public class Person {
+    /**
+     * Primary key.
```
Comment (01sas) [inline src/main/java/ru/smirnovv/person/Person.java]: The id of the person.

File: src/main/java/ru/smirnovv/person/Person.java [hunk 24]
```diff
@@ -0,0 +1,74 @@
...
+     */
+    @Id
+    @GeneratedValue
+    private Long id;
+
+    /**
+     * Person name.
```
Comment (01sas) [inline src/main/java/ru/smirnovv/person/Person.java]: The name of the person.

File: src/main/java/ru/smirnovv/person/Person.java [hunk 25]
```diff
@@ -0,0 +1,74 @@
...
+    private String name;
+
+    /**
+     * Constructs an instance of Person.
+     */
+    public Person() {
+    }
+
+    /**
+     * Constructs an instance of Person with the given name.
```
Comment (01sas) [inline src/main/java/ru/smirnovv/person/Person.java]: Constructs an instance with the specified properties.

File: src/main/java/ru/smirnovv/person/PersonController.java [hunk 26]
```diff
@@ -0,0 +1,122 @@
...
+import ru.smirnovv.ErrorType;
+
+import javax.servlet.http.HttpServletRequest;
+
+import static org.springframework.http.HttpStatus.BAD_REQUEST;
+import static org.springframework.util.Assert.notNull;
+import static ru.smirnovv.person.PersonService.checkName;
+
+/**
+ * The class handles requests associated with manages the persons.
```
Comment (01sas) [inline src/main/java/ru/smirnovv/person/PersonController.java]: A REST controller that serves registered in the system persons.

File: src/main/java/ru/smirnovv/person/PersonController.java [hunk 27]
```diff
@@ -0,0 +1,122 @@
...
+ */
+@RestController
+@RequestMapping("/person")
+public class PersonController {
+    /**
+     * Service that manages the persons.
```
Comment (01sas) [inline src/main/java/ru/smirnovv/person/PersonController.java]: A service that manages registered in the system persons.

File: src/main/java/ru/smirnovv/person/PersonController.java [hunk 28]
```diff
@@ -0,0 +1,122 @@
...
+     */
+    private final PersonService personService;
+
+    /**
+     * Constructs an instance with the given {@link PersonService}.
```
Comment (01sas) [inline src/main/java/ru/smirnovv/person/PersonController.java]: Constructs an instance with injected dependencies.

File: src/main/java/ru/smirnovv/person/PersonController.java [hunk 29]
```diff
@@ -0,0 +1,122 @@
...
+     *
+     * @param personService service that manages the persons.
+     * @throws IllegalArgumentException if 'personService' is null.
```
Comment (01sas) [inline src/main/java/ru/smirnovv/person/PersonController.java]: Not required.

File: src/main/java/ru/smirnovv/person/PersonController.java [hunk 30]
```diff
@@ -0,0 +1,122 @@
...
+        notNull(personService, "Argument 'personService' can not be null");
+
+        this.personService = personService;
+    }
+
+    /**
+     * Lists of persons.
+     *
+     * @param pageable information for pagination.
+     * @return page of persons.
```
Comment (01sas) [inline src/main/java/ru/smirnovv/person/PersonController.java]: the page of retrieved persons.

File: src/main/java/ru/smirnovv/person/PersonController.java [hunk 31]
```diff
@@ -0,0 +1,122 @@
...
+     */
+    @Autowired
+    public PersonController(final PersonService personService) {
+        notNull(personService, "Argument 'personService' can not be null");
+
+        this.personService = personService;
+    }
+
+    /**
+     * Lists of persons.
```
Comment (01sas) [inline src/main/java/ru/smirnovv/person/PersonController.java]: Lists all persons registered in the system.

File: src/main/java/ru/smirnovv/person/PersonController.java [hunk 32]
```diff
@@ -0,0 +1,122 @@
...
+     *
+     * @param pageable information for pagination.
```
Comment (01sas) [inline src/main/java/ru/smirnovv/person/PersonController.java]: a paging information.

File: src/main/java/ru/smirnovv/person/PersonController.java [hunk 33]
```diff
@@ -0,0 +1,122 @@
...
+     * @return page of persons.
+     */
+    @GetMapping
+    public final Page<Person> list(@PageableDefault(sort = "id") final Pageable pageable) {
+        return personService.list(pageable);
+    }
+
+    /**
+     * Adds the person if the name passed the verification.
```
Comment (01sas) [inline src/main/java/ru/smirnovv/person/PersonController.java]: Adds a new person.

File: src/main/java/ru/smirnovv/person/PersonController.java [hunk 34]
```diff
@@ -0,0 +1,122 @@
...
+    @PutMapping(consumes = MediaType.APPLICATION_FORM_URLENCODED_VALUE)
+    public final Person add(@RequestParam final String name) {
+        checkName(name);
+        return personService.add(name);
+    }
+
+    /**
+     * Returns the person by id if it exists.
+     *
+     * @param id id whose persons is to be found.
```
Comment (01sas) [inline src/main/java/ru/smirnovv/person/PersonController.java]: the id of the person.

**Pull request merged**